common/MigrationUtils.js

```diff
@@ -19,10 +27,76 @@ function shouldCommitDeployment(network) {
   );
 }
 
-// Helper function to deploy a contract and get the result.
-async function deployAndGet(deployer, contractType, ...args) {
+function isEmptyObject(obj) {
+  return Object.entries(obj).length === 0 && obj.constructor === Object;
+}
+
+// Extracts the transaction options from a transaction arg list. If the arg list doesn't have a transaction options
+// element, then it creates one and returns it.
+function getTxnOptions(argList) {
+  const emptyTxnOptions = {};
+  if (argList.length === 0) {
+    // No arguments, so add and return the empty txn options object.
+    argList.push(emptyTxnOptions);
+    return emptyTxnOptions;
+  } else {
+    // Determine if the last element is the txn options object.
+    const lastElement = argList[argList.length - 1];
+
+    if (
+      lastElement.from === undefined &&
```

Comment:
Obviously not your fault but this is bad API design, smh....

Reply:
Agreed :/